src/routes/ArticleDetailPage.jsx

```
@@ -1,49 +1,84 @@
 import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import Header from "../components/Header";
 import ArticleFinder from "../apis/ArticleFinder";
+import CommentList from "../components/CommentList";
 
 const ArticleDetailPage = () => {
   const { id } = useParams();
   const [selectedArticle, setSelectedArticle] = useState(null);
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const response = await ArticleFinder.get(`/articles/${id}`);
-        setSelectedArticle(response.data.article);
+        const articleResponsePromise = ArticleFinder.get(`/articles/${id}`);
+        const commentsResponsePromise = ArticleFinder.get(
+          `/articles/${id}/comments`
+        ).catch((error) => {
+          if (error.response && error.response.status === 404) {
+            return { data: { comments: [] } };
+          }
+          throw error;
```

We wouldn't want to throw an error here, because an error has already been caught in order to get to this line of code. It also doesn't look like there's anywhere ready to catch this error. We might consider handling this at this point with a state such as isError instead (that could cause an error message to pop up)